internal/cluster/refresh_status_preprocessor.go

```diff
@@ -47,24 +49,55 @@ func newRefreshPreprocessor(log logrus.FieldLogger, hostAPI host.API, operatorsA
 	}
 }
 
+func deserializeJSONList(jsonString string) ([]string, error) {
+	var list []string
+	if jsonString != "" {
+		err := json.Unmarshal([]byte(jsonString), &list)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return list, nil
+}
+
+func shouldIgnoreValidation(ignoredValidations []string, validationId string) bool {
+	if !common.MayIgnoreValidation(validationId, common.NonIgnorableClusterValidations) {
+		return false
+	}
+	if swag.ContainsStrings(ignoredValidations, "all") {
+		return true
+	}
+	return swag.ContainsStrings(ignoredValidations, validationId)
+}
+
 func (r *refreshPreprocessor) preprocess(ctx context.Context, c *clusterPreprocessContext) (map[string]bool, map[string][]ValidationResult, error) {
 	stateMachineInput := make(map[string]bool)
 	validationsOutput := make(map[string][]ValidationResult)
 	checkValidationsInStatuses := []string{
 		models.ClusterStatusInsufficient, models.ClusterStatusReady, models.ClusterStatusPendingForInput, models.ClusterStatusPreparingForInstallation,
 	}
+	var ignoredValidations []string
+	var err error
+	if c.cluster != nil && c.cluster.IgnoredClusterValidations != "" {
+		ignoredValidations, err = deserializeJSONList(c.cluster.IgnoredClusterValidations)
```

Comment:
Maybe be better to move it closer to usage?

Reply:
It seems that in each place you ignore err and just log it can we have a function that gets cluster and returns ignored validations? and log in case of error?